File: Array/941. Valid Mountain Array/Valid_Mountain_Array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        N = len(arr)
        i = 0
        
        ## first check edge cases, which means that the peak cannot be the first and the last element
        if N == 0:
            return False

            
        ## walk up
        while i + 1 < N and arr[i] < arr[i + 1]:
            i += 1
            
        ## check if the peak is the first or the last            
        if i == 0 or i == N - 1:
            return False
            
        ## walk down
        while i + 1 < N and arr[i] > arr[i + 1]:
            i += 1
            
        return i == N - 1
        

File: Array/941. Valid Mountain Array/Valid_Mountain_Array.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var validMountainArray = function(arr) {
    let N = arr.length;
    let i = 0;
    
    // edge case
    if(N === 0) return false;
    
    // walk up
    while(i + 1 < N && arr[i] < arr[i + 1]){
        i ++;
    }
    
    // check the peak cannot be the first nor the last
    if (i === 0 || i === N - 1) return false;
    
    // walk down
    while (i + 1 < N && arr[i] > arr[i + 1]){
        i ++;
    }

   return i === N - 1;

};

// Link: https://leetcode.com/problems/valid-mountain-array/
